### PR TITLE
Throw an error if the getSettings or validate promises take more that…

### PR DESCRIPTION
**Pull request by dompuiu**
… 10 seconds to return a result.

### CODE REVIEW

**Comment by Aaronius** [comment]
I would recommend putting the logic in the parent instead of the child. While it's good that we're checking that the promises that `getSettings` and `validate` resolve within a certain amount of time, it would be _better_ in my opinion to check that Lens actually receives the message within that amount of time.

**Comment by Aaronius** [comment]
This is why we do the "render timeout" in the parent instead of the child as well: https://github.com/Adobe-Marketing-Cloud/reactor-bridge/blob/c76cde0e6ae3b0feb323ee09929d1840fae92959/src/parent.js#L132-L135

**Comment by Aaronius** [comment]
I'd also recommend making it configurable by Lens like we did with the render timeout:

https://github.com/Adobe-Marketing-Cloud/reactor-bridge/blob/c76cde0e6ae3b0feb323ee09929d1840fae92959/src/parent.js#L80

So that we don't have to publish a new extension bridge version if we need to tweak it.

**Comment by dompuiu** [comment]
> If you want, you can remove the `clearTimeout` calls. I don't mind either way.

If I remove the clearTimeouts calls, won't that throw an error after 10 seconds?

**Comment by Aaronius** [comment]
Why do you think it might throw an error? Are you asking if it would reject the promise returned from `Promise.race()`? If so, then no, not as long as the other promise involved in the `Promise.race()` resolves first. Once a promise is resolved it can't get rejected and vice-versa. This is pretty simple to test in your console:

```
Promise.race([
  new Promise((resolve, reject) => {
    setTimeout(() => { reject('timed out'); }, 5000);
  }),
  new Promise((resolve) => {
    setTimeout(() => { resolve('validate returned'); }, 1000);
  })
])
.then((value) => { console.log('resolved', value); })
.catch((reason) => { console.log('rejected', reason) });
```

**Comment by dompuiu** [comment]
I've removed the `timeoutId` and the clearing of it.

File: src/parent.js [hunk 1]
```diff
@@ -115,8 +116,32 @@ export const loadIframe = options => {
...
+                validate: (...args) =>
+                  Promise.race([
+                    new Promise((_, reject) => {
+                      setTimeout(() => {
+                        reject(
+                          'A timeout occurred because the extension took longer than ' +
+                            PROMISE_TIMEOUT / 1000 +
+                            ' seconds to return the validation result.'
+                        );
+                      }, PROMISE_TIMEOUT);
```
Comment (Aaronius) [inline src/parent.js]: The timeout should be configurable by Lens.

**Comment by Aaronius** [comment]
Tests are also needed.

**Pull request merged**